### DISCUSSION TITLE
Any way to stop DisplayName truncating InlineData?

### DISCUSSION THREAD

**Discussion by VirtualTim** [xunit/xunit#2225] [Question] [answered]
When using InlineData it looks like the test name truncates each variable to 50 chars.

Example
```cpp
[Theory]
[InlineData(@"\\some\long\input\on\a\network\share\folder\a_file_input.png")]
public static void DoATest(string datasetPath) {
	//something to test
}
```
will generate something like:
```xml
<UnitTestResult testName="DoATest(datasetPath: "\\\\some\\long\\input\\on\\a\\network\\share\\fol"...)" />
```

In my specific use case I'm running tests against a number of files on a network share, and because of this most of the tests end up with the same name.
Is this limit configurable somewhere?

**Comment by VirtualTim** [comment]
As a side note, it looks like back slashes are being escaped when they don't need to be, but that's comparatively minor.

**Reply by bradwilson** [reply]
This is purposeful; we wanted the text there to be copy-pasteable into your source code.

**Reply by mwpowellhtx** [reply]
Do you mean, i.e. from `'\t'`, to `"\\t"` (?)

I'm not positive, but I think it is also destructive of the passed argument values themselves, at least based on one such result I was experiencing recently. Did not know what was happening, why the parser under test was rejecting the text. Until I realized it was not seeing the TAB character (i.e. WHITESPACE, was INTENTIONAL!) rather as the squashed two character string.

**Comment by bradwilson** [comment, marked as answer]
So, there are two thoughts, but the overriding message is: not today in v2.

First thought: we are adding in v3 support for setting display name (and other metadata) on a per data-row basis, which is one way that this problem could be solved (you can name the row in a way that helps you identify it, so you're not depending on our object serialization system to help that identification).

Second thought: I don't think we have a v3 feature for replacing this behavior (which today is covered by the [ArgumentFormatter class](https://github.com/xunit/assert.xunit/blob/main/Sdk/ArgumentFormatter.cs)). I'm open to ideas about how this might be accomplished, if the display name system described above is not suitable.

**Reply by VirtualTim** [reply]
Thanks for the reply.
I think the easiest (from a user point of view) solution would be to remove that hardcoded 50 chars, and allow that to be configured in the xunit.runner.json.

I guess as a work around I could split the arguments into multiple < 50 char arguments, but that's just going to lead to uglier test names. However the current behaviour is that half my tests have the same name, which breaks the Jenkins xunit reporting. Not to mention people viewing the report can't tell what test result links to each test.

Also I think this near the end of the file you linked is why backslashes get unnessicarily escaped:
```
if (ch == '\\') // backslash
	value = @"\\";
```

**Reply by meirkr** [reply]
What if as a WA you split the long path arg into 2 args: baswUrl and relativePath?
(Their order can be switched if it helps)

Another option
Hold the baseUrl as a single const definition or maybe in some config file.
This way if its location changes one day, you won't need to modify all your tests...

**Reply by VirtualTim** [reply]
Thanks meirkr.

> Hold the baseUrl as a single const definition or maybe in some config file.

That's what I'm actually already doing, I'm just concatenating the strings in the InlineData.

> What if as a WA you split the long path arg into 2 args: baswUrl and relativePath?

I thought about that, and it's a decent work-around. In my use case I'd have to use 3-4 arguments, which is a bit ugly.
Also since the test input is a file on a network share it's really handy to be able to just copy/paste that path if a test fails.
Also this is going to make my test names even longer...

I may have to end up doing this, it's just weird this is an issue.
I mean I get that there's a limit, but 50 chars seems really small, and it's a surprise it can't be configured. I'd assumed I missed something.
